src/eckit/log/Log.h

```diff
@@ -29,8 +30,7 @@ typedef void (*channel_callback_t)(void* data, const char* msg);
 /// Singleton holding global streams for logging
 ///
 
-class Log {
-
+class Log: private NonCopyable {
```

Comment:
Struct Log is essentially a namespace. If you look at the history of this code, it predates namespaces in C++...

I'm not sure if there would be knock-on impacts of changing it to be a namespace. There is a _lot_ of code which makes use of the log functionality. And (in any case) it would leave us out of our general conventions, in that we would have to have a namespace called "Log" rather than "log" or we would really break a _lot_ of stuff.

This PR already does far too much changing things (mostly for the better) at the same time as trying to fix an underlying issue. Makes it _much_ harder to see if/where it actually fixes the issue. I'm generally inclined towards separating 'cleanup' PRs and and functional PRs for the ease of correctness reviewing. So if we want to play with this, it should be in a separate PR.